src/components/Navbar/index.tsx

```diff
@@ -9,30 +9,99 @@ import CommentSummaryButton from '../CommentSummaryButton';
 import LoginButton from '../LoginButton';
 import { ApplicationState } from '../../reducers';
 import { Comment, selectVersionComments } from '../../reducers/comments';
-import { Version, selectCurrentVersionInfo } from '../../reducers/versions';
+import {
+  Version,
+  fetchVersion,
+  getVersionInfo,
+  selectCurrentVersionInfo,
+} from '../../reducers/versions';
 import { ConnectedReduxProps } from '../../configureStore';
 import { User, selectCurrentUser, requestLogOut } from '../../reducers/users';
 import styles from './styles.module.scss';
 
 export type PublicProps = {
+  _nextBaseVersionImprint?: string;
   _requestLogOut: typeof requestLogOut;
   reviewersHost: string;
 };
 
+export type DefaultProps = {
+  _fetchVersion: typeof fetchVersion;
+};
+
 type PropsFromState = {
   comments: Comment[] | undefined;
   user: User | null;
+  currentBaseVersion: Version | null | undefined | false;
+  currentBaseVersionId: number | undefined | false;
   currentVersion: Version | null | undefined | false;
 };
 
-type Props = PublicProps & PropsFromState & ConnectedReduxProps;
+type Props = PublicProps & DefaultProps & PropsFromState & ConnectedReduxProps;
+
+type State = {
+  nextBaseVersionImprint: string | undefined;
+};
 
-export class NavbarBase extends React.Component<Props> {
+export class NavbarBase extends React.Component<Props, State> {
   static defaultProps = {
+    _fetchVersion: fetchVersion,
     _requestLogOut: requestLogOut,
     reviewersHost: process.env.REACT_APP_REVIEWERS_HOST,
   };
 
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      nextBaseVersionImprint: props._nextBaseVersionImprint || undefined,
+    };
+  }
+
+  componentDidMount() {
+    this.loadData();
+  }
+
+  componentDidUpdate() {
+    this.loadData();
+  }
+
+  loadData() {
+    const {
+      _fetchVersion,
+      currentVersion,
+      currentBaseVersion,
+      currentBaseVersionId,
+      dispatch,
+    } = this.props;
+    const { nextBaseVersionImprint } = this.state;
+
+    if (
+      currentBaseVersion &&
+      nextBaseVersionImprint !== currentBaseVersion.version
```

Comment:
I know that this has been around pretty much forever, but looking at this code I am suddenly realizing that using `version` as the property name for the "version name/number" is actually quite confusing. A quick glance at this had me wondering why we were extracting a `version` from a `Version` object. I don't know if this is worth thinking about addressing in a follow-up by renaming this property.

Reply:
I agree. If you have a minute, I suggest filing a code quality issue to rename it to `versionString` or something.